s2s/time.py

```diff
@@ -170,6 +174,35 @@ def map_years(
 
         return index
 
+    def map_year_to_data(self, input_data: Union[pd.Series, pd.DataFrame, xr.Dataset, xr.DataArray],
+        flat: bool = False
+        ) -> pd.DataFrame:
+        """Map the calendar to input data period.
+        
+        Get time range from input data and generate correpodning interval index 
+        using ``map_year`` and ``map_years`` functions.
+
+        Args:
+            input_data: Input data for year mapping. Its index must be pandas.DatetimeIndex.
```

Comment:
year-mapping - are we only mapping the years, or are we also accounting for the exact start/end date of the data.

Reply:
Thanks for the suggestion. I rephrase it as "input_data: Input data for datetime mapping...", sounds much proper now.